include/picolibrary/circular_buffer.h

```diff
@@ -66,6 +72,246 @@ struct Without_Overflow_Underflow_Protection {
+     */
+    using Size = std::size_t;
+
+    /**
+     * \brief A reference to a circular buffer element.
+     */
+    using Reference = Value &;
+
+    /**
+     * \brief A reference to a const circular buffer element.
+     */
+    using Const_Reference = Value const &;
+
+    /**
+     * \brief Constructor.
+     */
+    Circular_Buffer() noexcept = default;
+
+    Circular_Buffer( Circular_Buffer && ) = delete;
+
+    Circular_Buffer( Circular_Buffer const & ) = delete;
+
+    /**
+     * \brief Destructor.
+     */
+    ~Circular_Buffer() noexcept
+    {
+        while ( not empty() ) { pop(); } // while
+    }
+
+    auto operator=( Circular_Buffer && ) = delete;
+
+    auto operator=( Circular_Buffer const & ) = delete;
+
+    /**
+     * \brief Access the first element in the circular buffer.
+     *
+     * \warning Calling this function on an empty circular buffer results in undefined
+     *          behavior.
+     *
+     * \return The first element in the circular buffer.
+     */
+    auto front() noexcept -> Reference
+    {
+        return *std::launder( reinterpret_cast<Value *>( m_read ) );
+    }
+
+    /**
+     * \brief Access the first element in the circular buffer.
+     *
+     * \warning Calling this function on an empty circular buffer results in undefined
+     *          behavior.
+     *
+     * \return The first element in the circular buffer.
+     */
+    auto front() const noexcept -> Const_Reference
+    {
+        return *std::launder( reinterpret_cast<Value const *>( m_read ) );
+    }
+
+    /**
+     * \brief Access the last element in the circular buffer.
+     *
+     * \warning Calling this function on an empty circular buffer results in undefined
+     *          behavior.
+     *
+     * \return The last element in the circular buffer.
+     */
+    auto back() noexcept -> Reference
+    {
+        return *std::launder( reinterpret_cast<Value *>(
+            m_write == &m_storage[ 0 ] ? &m_storage[ N - 1 ] : m_write - 1 ) );
+    }
+
+    /**
+     * \brief Access the last element in the circular buffer.
+     *
+     * \warning Calling this function on an empty circular buffer results in undefined
+     *          behavior.
+     *
+     * \return The last element in the circular buffer.
+     */
+    auto back() const noexcept -> Const_Reference
+    {
+        return *std::launder( reinterpret_cast<Value const *>(
+            m_write == &m_storage[ 0 ] ? &m_storage[ N - 1 ] : m_write - 1 ) );
+    }
+
+    /**
+     * \brief Check if the circular buffer is empty.
+     *
+     * \return true if the circular buffer is empty.
+     * \return false if the circular buffer is not empty.
+     */
+    auto empty() const noexcept
+    {
+        return not m_size;
+    }
+
+    /**
+     * \brief Check if the circular buffer is full.
+     *
+     * \return true if the circular buffer is empty.
+     * \return false if the circular buffer is not empty.
```

Comment:
Documentation is not correct.
```suggestion
     * \return true if the circular buffer is full.
     * \return false if the circular buffer is not full.
```